### DISCUSSION TITLE
Field validation rules are not being respected

### DISCUSSION THREAD

**Discussion by krazyjakee** [directus/directus#12175] [Archive]
I added an email validation boolean to the user collection so that I can allow certain privileges when using the service to those who have validated their emails.

On another collection, let's call it "Contacts", I have set a field validation rule so that the creating user must have validated their email. However, this rule is ignored.

![image](https://user-images.githubusercontent.com/623463/158264125-b78d9997-95c1-4253-8777-226b3f41b8fa.png)

How can I get this rule to be respected, or, how can I achieve locking down certain features to users with certain attributes?

Thanks.

### To Reproduce

- Add a custom field to the users object.
- Create a field validation rule on an object to prevent the user from doing it if that custom field is a certain value
- Test to see if the user can perform the task

### What version of Directus are you using?

9.6.0

### What version of Node.js are you using?

14.0.0

### What database are you using?

Postgres 12

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
I think this is a slight misunderstanding in what Field Validation is supposed to do. Field Validation controls what field values are allowed when the item is created, but does not check permissions to create said item. In this case, you're effectively telling Directus to only allow creating contacts where a nested user_created record is provided with a email_verified = true flag, or in code:

```
POST /contacts

{
  "user_created": {
    "email_verified": true
  }
}
```

However, what you're looking for is that you only want to allow user's that have the `email_verified` flag set to true to be able to create the item in the first place. Are you working with Directus Users, or is this a standalone users table?

**Reply by krazyjakee** [reply]
I am working with Directus users.

Thanks for the reply.

**Reply by rijkvanzanten** [reply]
I don't think you'll be able to configure a permissions rule that will check something against a field value where the field exists on the current user (the permissions currently allow for `field = $CURRENT_USER.field`, but not for `$CURRENT_USER.field = value`). You should be able to do this through a custom filter hook though 🙂
